### PR TITLE
*: Fix invalid memory access when shutting down

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/8408

Problem Summary:

There is an invalid memory access. This is because the logging system is already closed

image
image

### What is changed and how it works?

When `sub_root_of_query_storage_task_mem_trackers` and its children are destructoring, it means the process is shutting down and the logging system is stopped.

Suppress the logging of these memory trackers.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - Add a line after store_idnet is got to mock an exception thrown and tiflash stopped `throw Exception("Mock exception thrown");`
https://github.com/pingcap/tiflash/blob/3d3a5f14d51a55c636223e1ac9c9038e082ebb62/dbms/src/Server/Server.cpp#L1707-L1716
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Common/MemoryTracker.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include <common/types.h>
 
 #include <atomic>
+#include <boost/noncopyable.hpp>
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Common/MemoryTracker.h:21]: it is need by `struct TemporarilyDisableMemoryTracker : private boost::noncopyable` in line 192

**Pull request merged**